tools/seal_oem_partition.go

```diff
@@ -0,0 +1,204 @@
+package tools
+
+import (
+	"bytes"
+	"cos-customizer/tools/partutil"
+	"fmt"
+	"io/ioutil"
+	"log"
+	"os"
+	"os/exec"
+	"strconv"
+	"strings"
+)
+
+// SealOEMPartition sets the hashtree of the OEM partition
+// with "veritysetup" and modifies the kernel command line to
+// verify the OEM partition at boot time.
+func SealOEMPartition(oemFSSize4K uint64) error {
+	const devName = "oemroot"
+	const veritysetupImgPath = "./veritysetup.img"
+	imageID, err := loadVeritysetupImage(veritysetupImgPath)
+	if err != nil {
+		return fmt.Errorf("cannot load veritysetup image at %q, error msg:(%v)", veritysetupImgPath, err)
+	}
+	log.Println("docker image for veritysetup loaded.")
+	if err := unmountOEMPartition(); err != nil {
+		return fmt.Errorf("cannot umount OEM partition, error msg:(%v)", err)
+	}
+	log.Println("OEM parititon unmounted.")
+	hash, salt, err := veritysetup(imageID, oemFSSize4K)
+	if err != nil {
+		return fmt.Errorf("cannot run veritysetup, input:oemFSSize4K=%d, "+
+			"error msg:(%v)", oemFSSize4K, err)
+	}
+	grubPath, err := mountEFIPartition()
+	log.Println("EFI parititon mounted.")
+	if err != nil {
+		return fmt.Errorf("cannot mount EFI partition (/dev/sda12), error msg:(%v)", err)
+	}
+	partUUID, err := partutil.GetPartUUID("/dev/sda8")
+	if err != nil {
+		return fmt.Errorf("cannot read partUUID of /dev/sda8")
+	}
+	if err := appendDMEntryToGRUB(grubPath, devName, partUUID, hash, salt, oemFSSize4K); err != nil {
+		return fmt.Errorf("error in appending entry to grub.cfg, input:oemFSSize4K=%d, "+
+			"error msg:(%v)", oemFSSize4K, err)
+	}
+	log.Println("kernel command line modified.")
+	if err := removeVeritysetupImage(imageID); err != nil {
+		return fmt.Errorf("cannot remove veritysetup container or image, error msg:(%v)", err)
```

Comment:
Maybe this should not be error and can be a warning as everything was done by the container?

Reply:
If it fails to delete the image, the test expecting only one ubuntu image will fail

Reply:
Error makes sense here, since we don't want to leak the image onto the user's images/VMs.